ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/utils/StringUtils.java

```diff
@@ -172,4 +174,28 @@ public static String getErrorStringValue(Throwable error) {
     public static String convertUnicode(char character) {
         return "\\u{" + Integer.toHexString((int) character) + "}";
     }
+
+    /**
+     * Replace ballerina unicode format codes with their unicode character
+     * for a given string.
+     *
+     * @param toConvert String need to convert.
+     * @return Reformatted string.
+     */
+    public static String convertUnicodeToCharacter(String toConvert) {
+        Matcher matcher = Pattern.compile("\\\\u\\{([\\da-fA-F]*)}").matcher(toConvert);
+
+        StringBuilder stringBuilder = new StringBuilder();
+        int currentPosition = 0;
+        while (matcher.find()) {
+            stringBuilder.append(toConvert, currentPosition, matcher.start());
+            int code = Integer.parseInt(matcher.group(1), 16);
+            char[] chars = Character.toChars(code);
+            stringBuilder.append(chars);
```

Comment:
Shall we reduce this to a one line and avoid declaring `chars` if we are not using it later ? 
```suggestion
            stringBuilder.append(Character.toChars(code);
```

Reply:
Updated